crates/testing/src/task_helpers.rs

```diff
@@ -148,15 +235,67 @@ async fn build_quorum_proposal_and_signature(
         timestamp: 0,
         proposer_id: *api.public_key(),
     };
-    let signature = <BLSPubKey as SignatureKey>::sign(private_key, leaf.commit().as_ref())
+
+    let mut signature = <BLSPubKey as SignatureKey>::sign(private_key, leaf.commit().as_ref())
         .expect("Failed to sign leaf commitment!");
-    let proposal = QuorumProposal::<TestTypes> {
-        block_header,
-        view_number: ViewNumber::new(view),
+    let mut proposal = QuorumProposal::<TestTypes> {
+        block_header: block_header.clone(),
+        view_number: ViewNumber::new(1),
         justify_qc: QuorumCertificate::genesis(),
         timeout_certificate: None,
         proposer_id: leaf.proposer_id,
     };
+    for cur_view in 2..=view {
+        consensus.state_map.insert(
+            ViewNumber::new(1),
+            View {
+                view_inner: ViewInner::Leaf {
+                    leaf: leaf.commit(),
+                },
+            },
+        );
+        consensus.saved_leaves.insert(leaf.commit(), leaf.clone());
+        let quorum_membership = handle.hotshot.inner.memberships.quorum_membership.clone();
+        let quorum_data = QuorumData {
+            leaf_commit: leaf.commit(),
+        };
+        let created_qc = build_cert::<
+            TestTypes,
+            QuorumData<TestTypes>,
+            QuorumVote<TestTypes>,
+            QuorumCertificate<TestTypes>,
+        >(
+            quorum_data,
+            &quorum_membership,
+            ViewNumber::new(cur_view - 1),
+            public_key,
+            private_key,
+        );
+        let parent_leaf = leaf.clone();
+        let leaf_new_view = Leaf {
+            view_number: ViewNumber::new(cur_view),
+            justify_qc: created_qc.clone(),
+            parent_commitment: parent_leaf.commit(),
+            block_header: block_header.clone(),
+            block_payload: None,
+            rejected: vec![],
+            timestamp: 0,
+            proposer_id: *api.public_key(),
```

Comment:
I think this should be the leaders public key, we can get it from the membership by calling `get_leader(curr_view)`

Reply:
Good catch, never really think about view 3 T.T  I'll add one comment for this.